### PR TITLE
Environment.js rewrite

### PR DESCRIPTION
**Pull request by ashwell**
The environment is now inferred on bootstrap of the application and is not mutable once set. A restart of the application will be required to switch environments. The current environment will be inferred from the hostname with the subdomain ignored, unless the environment is set in a query param or cookie.

incrementing a major version since the `setEnvironment` function is now gone

### CODE REVIEW

**Comment by justynaw** [comment]
Looks good. Merge if you tested it.
One question - you said you removed the changing of environment from the bottom right menu - is this a separate change?

**Comment by ashwell** [comment]
@justynaw yeah, that change is in the dashboard [PR here](https://github.com/Enplug/dashboard/pull/48) I forgot to assign you to that one

File: dist/dashboard-utils.js [hunk 1]
```diff
@@ -1490,117 +1490,208 @@ angular.module('enplug.utils').factory('EndpointOptions', ['$log', 'Environment'
...
+    // set 'fallback' environment enum here for easy changing later
+    fallbackEnv = this.STAGING;
+
+    // maps domains to expected environment
+    // todo remove these notes below (maybe?)
+    //  prod dashboard is on .com
+    //  prod apps are on .net
+    //  staging apps and dashboard are on .in
+    //  local is for local dev envs (must be set in your computers "hosts" file to 127.0.0.1)
+    domainEnvMap = Object.assign( Object.create( null ), {
```
Comment (aleross) [inline dist/dashboard-utils.js]: I happened to still be watching this repository so I saw this PR (I'm unsubscribing now), and as I was skimming through it the call to `Object.assign()` stood out. I don't think this is supported in any versions of IE: [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign).

Best of luck!

Comment (ashwell) [inline dist/dashboard-utils.js]: oh man, good catch THANKS! I'm still getting used to not being in a babel toolchain, I'll switch it to use an angular equivalent

**Pull request merged**